bob-common/src/mapper.rs

```diff
@@ -131,18 +125,38 @@ impl Virtual {
         trace!("extract indexes of target nodes");
         trace!("nodes available: {}", self.nodes.len());
         let offset = self.support_nodes_offset.fetch_add(1, Ordering::Relaxed);
-        let mut support_nodes = Vec::with_capacity(count);
-        for (id, node) in self.nodes.iter().skip(offset % self.nodes.len()) {
-            if target_nodes.iter().all(|i| i.index() != *id) {
+        let mut support_nodes: Vec<&Node> = Vec::with_capacity(count);
+        let mut index = 0;
+        let mut offset_counter = offset % self.nodes.len();
+        let mut available_count = 0;
+        while index < self.nodes.len() && offset_counter > 0 {
+            if self.nodes[index].connection_available() {
+                available_count += 1;
+                offset_counter -= 1;
+            }
+            index += 1;
+        }
+        let actual_offset = if index == self.nodes.len() {
+            offset % available_count
```

Comment:
This is offset within available nodes. It is incorrect to use it as offset within the whole `nodes` collection

Reply:
That's why I introduced additional calls to `Find` within loop in my example